faunadb-scala/src/main/scala/faunadb/errors/FaunaException.scala

```diff
@@ -4,7 +4,7 @@ import faunadb.{QueryError, QueryErrorResponse}
 
 object FaunaException {
   private[errors] def respToError(response: QueryErrorResponse) = {
-    response.errors.map(r => r.code + ": " + r.description).mkString(", ")
+    response.errors.iterator.map(r => r.code + ": " + r.description).mkString(", ")
```

Comment:
I see a microbenchmark coming... 😄

Reply:
I had very good experiences in the past fusing collection operations via `Iterators`.
IMHO using naively the Scala collection API is good way to allocate a lot :smile: 

Moreover, in that case we are folding over it using `mkString` to produce a `String`, any intermediate collections to get there are completely unnecessary.

Are those points convincing enough?

Reply:
Also this is the code path for error reporting, I am fine removing the change as it should not have a big impact anyway :)

Reply:
😄 Yeah, I agree with your points! :)

The only "downside" I could see of using `.iterator` there is that someone else looking at the code might think we used it because we were having some performance troubles in that part of the code. But that's fine, let's got with it!

It might be worthy to add this kind of things (e.g., "Always convert to Iterator first, if traversing a Collection more than once") to a list of best practices so that they are not a surprise to newcomers. There are some "best practices" we're doing which I think are not documented anywhere, like always making case classes final (which personally I'm not a big fan, btw 😄)

Probably, the closest thing we have now is our [Style Guide](https://faunadb.atlassian.net/wiki/spaces/CORE/pages/501612550/Scala+Style+Guide) page in Confluence? But, I have the feeling those might fit better into a new "Scala Best Practices" page...

@nathaniel-may, might have some thoughts about this! :)

Reply:
The only caveat with that rule is that if you're returning an collection, then you need to be careful to turn it into something non-lazy at the end, or you may wind up with unexpected behavior when your transformations do not run in the body of the method, but some time later. 

It's a good thing to point out though! I have a bad habit of reverting to foreach and a builder, but `.iterator` followed by `toVector` `toMap` etc is probably just as good or close enough.